app/models/fe/concerns/answer_sheet_concern.rb

```diff
@@ -83,6 +84,7 @@ def percent_complete(required_only = true)
 
       # determine which questions should count towards the questions total in the percent calculation
       countable_questions = question_sheets.collect{ |qs| qs.all_elements.questions }.flatten
+      countable_questions.select!{ |e| (restrict_to_pages & e.pages_on).any? } if restrict_to_pages.present?
       countable_questions.reject!{ |e| e.hidden?(self, elements_to_pages[e.id]) }
       countable_questions.reject!{ |e| !e.required } if required_only
```

Comment:
Not this patch, but: this could have been `.select!{ |e| e.required } if required_only`